src/omit.test.ts

```diff
@@ -12,13 +12,31 @@ describe('data first', () => {
     const result = omit(obj, ['a']);
     expect(result).toEqual({});
   });
+
+  test('non existing prop', () => {
+    // @ts-expect-error -- should not allow non existing props
+    const result = omit({ a: 1, b: 2, c: 3, d: 4 }, ['not', 'in'] as const);
+    expect(result).toEqual({ a: 1, b: 2, c: 3, d: 4 });
+  });
```

Comment:
1. In most of our new tests i've split type tests from regular tests, that way you can focus on the thing you are changing when running tests.

Can you create a new top-level "describe" category for "typing" and put the tests there?

2. You also don't need to *also* test the runtime behavior, as those are 2 different concerns of the test. Here for example there's no point in testing the runtime behavior because the typing error would prevent this code from running.

3. Add the specific typescript error (tsXXXX) we are suppressing here, so that if typescript make changes and that error stops firing, we can debug what the original error was (similar to how you'd add the error message to a `expect(...).toThrow(<ERROR_MESSAGE>)` instead of just `expect(...).toThrow()`